spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigMapConfigProperties.java

```diff
@@ -123,14 +145,59 @@ public void setNamespace(String namespace) {
 			this.namespace = namespace;
 		}
 
+		public Boolean isUseNameAsPrefix() {
+			return useNameAsPrefix;
+		}
+
+		public void setUseNameAsPrefix(Boolean useNameAsPrefix) {
+			this.useNameAsPrefix = useNameAsPrefix;
+		}
+
+		public String getExplicitPrefix() {
+			return explicitPrefix;
+		}
+
+		public void setExplicitPrefix(String explicitPrefix) {
+			this.explicitPrefix = explicitPrefix;
+		}
+
 		public boolean isEmpty() {
 			return !StringUtils.hasLength(this.name) && !StringUtils.hasLength(this.namespace);
 		}
 
+		// not used, but not removed because of potential compatibility reasons
 		public NormalizedSource normalize(String defaultName, String defaultNamespace) {
 			String normalizedName = StringUtils.hasLength(this.name) ? this.name : defaultName;
 			String normalizedNamespace = StringUtils.hasLength(this.namespace) ? this.namespace : defaultNamespace;
-			return new NormalizedSource(normalizedName, normalizedNamespace);
+			return new NormalizedSource(normalizedName, normalizedNamespace, null);
+		}
+
+		public NormalizedSource normalize(String defaultName, String defaultNamespace, boolean defaultUseNameAsPrefix,
+				String explicitPrefix) {
+			String normalizedName = StringUtils.hasLength(this.name) ? this.name : defaultName;
+			String normalizedNamespace = StringUtils.hasLength(this.namespace) ? this.namespace : defaultNamespace;
+
+			// if explicitPrefix is set, it takes priority over useNameAsPrefix
+			// (either the one from 'spring.cloud.kubernetes.config' or
+			// 'spring.cloud.kubernetes.config.sources')
+			if (StringUtils.hasText(explicitPrefix)) {
+				return new NormalizedSource(normalizedName, normalizedNamespace, explicitPrefix);
+			}
+
+			// useNameAsPrefix is a java.lang.Boolean and if it's != null, users have
+			// specified it explicitly
+			if (this.useNameAsPrefix != null) {
```

Comment:
this just shows that `spring.cloud.kubernetes.config.sources.useNameAsPrefix` has a higher order.